hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -508,7 +512,7 @@ private long tryReserveForUpdate(Data key) {
     }
 
     private boolean cachingAllowedFor(Data key) {
-        return cacheLocalEntries || !isOwn(key);
+        return cacheLocalEntries || localMember.isLiteMember() || !isOwn(key);
```

Comment:
The `getNodeEngine()` seems to have some overhead of a lifecycle check. Also `getLocalMember()` is redirected several times. The `ClusterService` seems to be the source of truth, so I picked `clusterService.getLocalMember().isLiteMember()` and cache the `ClusterService` as field. I hope this is the most performant and correct solution :)